woocommerce/woocommerce
# Run API tests with regular run and add script

### Changes proposed in this Pull Request:

This PR introduces a script like `wc-e2e.sh` for the API testing package. Additionally, this script has been added to be run in the nightly and release GitHub Workflows.

Closes https://github.com/woocommerce/woocommerce/issues/30656

Note that this bumps the Jest version down in the API Core Tests package as otherwise the script would fail to run with `Cannot destructure property 'config' of 'cacheKeyOptions' as it is undefined`.

### How to test the changes in this Pull Request:

1. Verify the CI passes
2. Run the new script and verify the tests work as expected:
```shell
BASE_URL=http://localhost:8084 USER_KEY=admin USER_SECRET=password npx wc-api-tests test:api 
```
3. Test the other scripts and verify they work as expected:
```shell
BASE_URL=http://localhost:8084 USER_KEY=admin USER_SECRET=password npx wc-api-tests test:hello 
```
```shell
BASE_URL=http://localhost:8084 USER_KEY=admin USER_SECRET=password npx wc-api-tests make:collection 
```
